crawler.py

```diff
@@ -756,6 +766,23 @@ def save(self, data, name='results.json'):
                 data, f, indent=2, sort_keys=True, separators=(',', ': '))
         self.logger.info("Saved data to %s", name)
 
+    def handle_clicking(self, domain):
+        # fetch the clickable elements on given domain
+        clickables = self.driver.find_elements_by_tag_name('a') + self.driver.find_elements_by_tag_name('button')
+
+        # strip the domain down so we can catch first party subdomains too
+        pattern = re.compile(r"https?://(www\.)?")
+        stripped_domain = pattern.sub('', domain).strip().strip('/')
+
+        # iterate through each found first party clickable on domain, try to interact with it
+        for clickable in clickables:
+            try:
+                if stripped_domain in clickable.get_attribute('href'):
```

Comment:
You could also do this filtering more directly with `find_element_by_css_selector` with [CSS attribute selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors). For example, `a[href^='https://google.com/']`: all links pointing to https URLs starting with "google.com/".

Reply:
Should make sure we don't click on an essentially identical href to the site you are on. No point in clicking on `https://www.facebook.com/#` or `https://www.facebook.com/` on `facebook.com`, for example. (We do now though.)

Reply:
This clicks links in a loop, which doesn't actually visit every link. Only the first one (?) wins.